src/NuGet.Core/NuGet.Packaging.Extraction/NuGet.Packaging.Extraction.csproj

```diff
@@ -0,0 +1,360 @@
+﻿<Project>
+  <PropertyGroup>
+    <RequiresSigningXplatAPIs>true</RequiresSigningXplatAPIs>
+  </PropertyGroup>
+
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), 'README.md'))\build\common.props" />
+  <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
+
+  <PropertyGroup>
+    <Description>NuGet's understanding of packages. Reading nuspec, nupkgs and package signing.</Description>
+    <TargetFrameworks>net45</TargetFrameworks>
+    <TargetFramework />
+    <NoWarn>$(NoWarn);CS1591;CS1574;CS1573;CS1572</NoWarn>
+    <NoWarn Condition=" $(TargetFramework.StartsWith('netstandard')) OR $(TargetFramework.StartsWith('netcoreapp'))">$(NoWarn);CS0414</NoWarn>
+    <PackProject>true</PackProject>
+    <Shipping>true</Shipping>
+    <IncludeInVSIX>false</IncludeInVSIX>
+    <XPLATProject>true</XPLATProject>
+    <UsePublicApiAnalyzer>false</UsePublicApiAnalyzer>
+    <DefineConstants>$(DefineConstants);IS_DESKTOP;IS_SIGNING_SUPPORTED</DefineConstants>
+    <IsDesktop>true</IsDesktop>
+  </PropertyGroup>
+
+  <PropertyGroup Condition=" '$(IsVsixBuild)' == 'true' ">
+    <TargetFrameworks />
+    <TargetFramework>$(NETFXTargetFramework)</TargetFramework>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+  </PropertyGroup>
+
+<ItemGroup>
+  <ProjectReference Include="..\NuGet.Configuration\NUGet.configuration.csproj" />
+  <ProjectReference Include="..\NuGet.versioning\NUGet.versioning.csproj" />
+</ItemGroup>
+  <!-- <ItemGroup Condition="'$(TargetFramework)' == 'net462'">
+    <PackageReference Include="NuGet.Configuration" Version="4.9.*" />
+    <PackageReference Include="NuGet.Versioning" Version="4.9.*" />
+  </ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)' == 'net45'">
+    <PackageReference Include="NuGet.Configuration" Version="4.5.*" />
+    <PackageReference Include="NuGet.Versioning" Version="4.5.*" />
+  </ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)' == 'net472'">
+    <PackageReference Include="NuGet.Configuration" Version="5.*-*" />
+    <PackageReference Include="NuGet.Versioning" Version="5.*-*" />
+  </ItemGroup> -->
+
+  <ItemGroup Condition=" '$(IsDesktop)' == 'true' ">
+    <Reference Include="System.Xml" />
+    <Reference Include="System.Xml.Linq" />
+    <Reference Include="System.IO.Compression" />
+    <Reference Include="System.Security" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <EmbeddedResource Include="compiler\resources\*" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Compile Remove="NuGet.Common\**" />
+    <Compile Remove="NuGet.Configuration\**" />
+    <EmbeddedResource Remove="NuGet.Common\**" />
+    <EmbeddedResource Remove="NuGet.Configuration\**" />
+    <None Remove="NuGet.Common\**" />
+    <None Remove="NuGet.Configuration\**" />
+  </ItemGroup>
+
+  <ItemGroup>
```

Comment:
Instead of compiling NuGet.Packaging, can we add a `net45` to the required project and add it as project references?

Reply:
i don't understand this feedback.
are you suggesting...that we don't have a new dll...and just add net45 to nuget.packaging?
or something else?

Reply:
@dominoFire and i discussed the purpose of this dll in a teams call. resolving this conversation.

Reply:
I would like to ask this question, may I know why creating NuGet.Packing.Extraction targeting net45 instead of adding net45 to nuget.packaging?

Reply:
our customer only needs extraction support. so i created this DLL with PackageExtractor.cs and all the classes that it needed to compile.
This allows us to do fewer ifdefs in the nuget.packaging codebase.

for example, what if signing needed net46, but i was trying to compile it for net45.
we could ifdef out classes or class members to make it compile...

i would be willing to take the net45 approach across the real (nuget.packaging) dll... if you and @dtivel and others think it is a better approach...but would likely defer to after merging this.

Reply:
@heng-liu  - happy to discuss "in person" if that would be better.

Reply:
Not sure if this would work: disable signing APIs in NuGet.Packaging for net45, by adding #if IS_SIGNING_SUPPORTED && !net45
But I don't how to add tests later on for net45,  if signing API is disabled..As the verification tests need the signing APIs to be enabled.

Reply:
will look into that idea. thx.

Reply:
tracking that possibility in this follow up issue: https://github.com/nuget/client.engineering/issues/461